### PR TITLE
python311Packages.connexion: 2.14.2 -> 3.0.1

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

Updated to provide compatibility with flask 2.3.x.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Not a valid fix for airflow. Requires a flask downgrade unfortunately.


Result of `nixpkgs-review pr 262544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.connexion</li>
    <li>python310Packages.connexion.dist</li>
    <li>python310Packages.py-swagger-ui</li>
    <li>python310Packages.py-swagger-ui.dist</li>
    <li>python311Packages.connexion</li>
    <li>python311Packages.connexion.dist</li>
    <li>python311Packages.py-swagger-ui</li>
    <li>python311Packages.py-swagger-ui.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/connexion/default.nix [hunk 1]
```diff
@@ -1,70 +1,98 @@
...
+, pytest-aiohttp
+, pytestCheckHook
 , testfixtures
 }:
 
 buildPythonPackage rec {
   pname = "connexion";
-  version = "2.14.2";
-  format = "setuptools";
+  version = "3.0.0a7";
```
Comment (jonringer) [inline pkgs/development/python-modules/connexion/default.nix:38]: seems that there is now a stable 3.0 release

**Review by jonringer — APPROVED** [review]
I think this can be merged into staging-next fine.

File: pkgs/development/python-modules/connexion/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/connexion/default.nix]: Updated, thanks!

**Comment by mweinelt** [comment]
> I think this can be merged into staging-next fine.

Master should work just fine, flask 2.3.3 is already there.

**Comment by mweinelt** [comment]
> apache-airflow> ModuleNotFoundError: No module named 'connexion.decorators.validation'

What is needed is a flask pin on airflow.

Result of `nixpkgs-review pr 262544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.connexion</li>
    <li>python310Packages.connexion.dist</li>
    <li>python310Packages.py-swagger-ui</li>
    <li>python310Packages.py-swagger-ui.dist</li>
    <li>python311Packages.connexion</li>
    <li>python311Packages.connexion.dist</li>
    <li>python311Packages.py-swagger-ui</li>
    <li>python311Packages.py-swagger-ui.dist</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 262544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.connexion</li>
    <li>python310Packages.connexion.dist</li>
    <li>python310Packages.py-swagger-ui</li>
    <li>python310Packages.py-swagger-ui.dist</li>
    <li>python311Packages.connexion</li>
    <li>python311Packages.connexion.dist</li>
    <li>python311Packages.py-swagger-ui</li>
    <li>python311Packages.py-swagger-ui.dist</li>
  </ul>
</details>

**Comment by boltzmannrain** [comment]
Tried this and I think apache-airflow is hitting https://github.com/apache/airflow/issues/35234 which doesn't yet have a fix
```
  File "/nix/store/x6dp50lbvz2b1av3rrvam5iyc5nlmwg5-python3.11-apache-airflow-2.7.1/lib/python3.11/site-packages/airflow/migrations/versions/0074_2_0_0_resource_based_permissions.py", line 30, in <module>
    from airflow.www.app import cached_app
  File "/nix/store/x6dp50lbvz2b1av3rrvam5iyc5nlmwg5-python3.11-apache-airflow-2.7.1/lib/python3.11/site-packages/airflow/www/app.py", line 50, in <module>
    from airflow.www.extensions.init_views import (
  File "/nix/store/x6dp50lbvz2b1av3rrvam5iyc5nlmwg5-python3.11-apache-airflow-2.7.1/lib/python3.11/site-packages/airflow/www/extensions/init_views.py", line 25, in <module>
    from connexion.decorators.validation import RequestBodyValidator
ModuleNotFoundError: No module named 'connexion.decorators.validation'
```

**Comment by mweinelt** [comment]
Yeah, apache-airflow is broken right now, because it also doesn't like flask 2.3.

**Comment by mweinelt** [comment]
Superseded by #271586.

**Pull request closed without merging**